### PR TITLE
Fixes kubernetes#217

### PR DESCRIPTION
**Pull request by jiangyuzhao**
If name has prefix 'func (', just return types.Name without parsing more. Fixes kubernetes#217

### CODE REVIEW

**Comment by jiangyuzhao** [comment]
@sttts Cloud you please also take a look?

**Comment by jiangyuzhao** [comment]
@lavalamp @sttts Cloud you pls take a look?

**Comment by lavalamp** [comment]
Also, you'll need to sign the CLA.

**Comment by jiangyuzhao** [comment]
@lavalamp I have signed the CLA and modify the test file. In the old implement it will parse the type error and treat it as package name.

**Comment by jiangyuzhao** [comment]
@lavalamp  Cloud you pls take a look?

**Comment by jiangyuzhao** [comment]
@lavalamp hello?

File: parser/parse_test.go [hunk 1]
```diff
@@ -62,6 +62,10 @@ func TestRecursive(t *testing.T) {
 	if foundC {
 		t.Error("Did not expect to find package c")
 	}
+	// only has three package: package "a", package "b", and package "" for all
+	if len(findTypes) != 3 {
```
Comment (lavalamp) [inline parser/parse_test.go:66]: I was expecting a check that the type identified is actually correct? From my reading the wrong code was still making a type, it's just not the right construction.

**Comment by lavalamp** [comment]
BTW we are in extremely different time zones, and I do this during working hours, so you can't expect a fast response, sorry

**Comment by jiangyuzhao** [comment]
@lavalamp Sorry for disturbing you at the weekend, I just thought it's daytime and forgot it's also weekend.  
Now I update the commit for type check, but it's somehow ugly code. I just found that if I check the length of `findTypes` which's keys are package names is ok, in the wrong code it will get keys like `func (*k8s.io/gengo/testdata/a.AA).AFunc(i *int, j int) (*k8s.io/gengo/testdata/a.A, k8s.io/gengo/te` which I don't know how it's parsed.

**Comment by lavalamp** [comment]
OK, it seems possible that this could break (in a different way) and the test could still pass, but this is probably good enough for now.

/lgtm
/approve

**Pull request merged**